### PR TITLE
Add Xcode as dependency of flutter_view_macos__start_up in .ci.yaml

### PR DESCRIPTION
**Pull request by a-wallen**
`flutter_view_macos__start_up` test landed in https://github.com/flutter/flutter/pull/111326, but lacked the dependencies that it needed to run successfully in postsubmit. This change adds Xcode as a dependency so that the `flutter_view_macos__start_up` test can be run in postsubmit.

There's no filed issue for this.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
> only affect code inside the .github directory, .cirrus.yml or .ci.yaml config files.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by a-wallen** [comment]
image

Tested by removing bringup in [`f4701f8` (#111624)](https://github.com/flutter/flutter/pull/111624/commits/f4701f8b0b3371315e53cf5bb11837006c166223).

File: .ci.yaml [hunk 1]
```diff
@@ -2508,6 +2508,10 @@ targets:
     recipe: devicelab/devicelab_drone
     timeout: 60
     properties:
+      dependencies: >-
+        [
+          {"dependency": "xcode", "version": "13f17a"}
```
Comment (keyonghan) [inline .ci.yaml:2513]: Shall this be `14a5294e`? All xcode has been bumped to version14.

**Comment by a-wallen** [comment]
image

Works with updated Xcode dependency `14a5294e`. Thanks for letting me know @keyonghan!

**Comment by keyonghan** [comment]
> Works with updated Xcode dependency 14a5294e. Thanks for letting me know @keyonghan!

Seems one more place referencing the old xcode: Mac flutter_gallery_macos__start_up. Could you fix that as well?

**Comment by a-wallen** [comment]
> Seems one more place referencing the old xcode: Mac flutter_gallery_macos__start_up. Could you fix that as well?

✅ done

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**